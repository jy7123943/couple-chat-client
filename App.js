import React, { Component } from 'react';
import { YellowBox } from 'react-native';
import { createAppContainer, createSwitchNavigator } from 'react-navigation';
import { AppLoading } from 'expo';
import * as Font from 'expo-font';
import { Ionicons } from '@expo/vector-icons';
import AuthNavigator from './src/screen/auth/AuthNavigator';
import io from 'socket.io-client';
import getEnvVars from './environment';
const { apiUrl } = getEnvVars();

YellowBox.ignoreWarnings(['Remote debugger']);
YellowBox.ignoreWarnings([
    'Unrecognized WebSocket connection option(s) `agent`, `perMessageDeflate`, `pfx`, `key`, `passphrase`, `cert`, `ca`, `ciphers`, `rejectUnauthorized`. Did you mean to put these under `headers`?'
]);
const socketConfig = {
  'force new connection' : true,
  'reconnection': true,
  'reconnectionDelay': 2000,
  'reconnectionDelayMax' : 60000,
  'reconnectionAttempts': 'Infinity',
  'timeout' : 10000,     
  transports: ['websocket'],
  upgrade: false
};
const socket = io(apiUrl, socketConfig);
export default class Root extends Component {
  constructor(props) {
    super(props);
    this.state = {
      isReady: false,
      userInfo: null,
      roomInfo: null
    };
  }

  setUserInfo = (userInfo) => {
    this.setState({
      ...this.state,
      userInfo
    });
  };

  setRoomInfo = (roomInfo) => {
    this.setState({
      ...this.state,
      roomInfo
    });
  };

  async componentDidMount() {
    await Font.loadAsync({
      Roboto: require('native-base/Fonts/Roboto.ttf'),
      Roboto_medium: require('native-base/Fonts/Roboto_medium.ttf'),
      ...Ionicons.font,
    });
    this.setState({ isReady: true });
  }

  render() {
    if (!this.state.isReady) {
      return <AppLoading />;
    }

    return (
      <AuthNavigator
        screenProps={{
          ...this.state,
          socket,
          setRoomInfo: this.setRoomInfo,
          setUserInfo: this.setUserInfo
        }}
      />
    );
  }
}
