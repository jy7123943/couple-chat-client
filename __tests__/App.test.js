import React from 'react';
import App from '../App';

import { shallow } from 'enzyme';

describe('<App />', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<App />);
  });

  it('matches snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  })

  it('has default states', () => {
    expect(wrapper.state().isReady).toBe(false);
    expect(wrapper.state().userInfo).toBe(null);
    expect(wrapper.state().roomInfo).toBe(null);
  });
});
