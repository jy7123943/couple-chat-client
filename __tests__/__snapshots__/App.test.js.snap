// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> matches snapshot 1`] = `ShallowWrapper {}`;

exports[`App.js matches snapshot 1`] = `ShallowWrapper {}`;

exports[`App.js snapshot 1`] = `ShallowWrapper {}`;

exports[`renders correctly 1`] = `ShallowWrapper {}`;
